pkg/cableengine/healthchecker/pinger.go

```diff
@@ -109,10 +112,12 @@ func (p *pingerInfo) Start() {
 			case <-p.stopCh:
 				return
 			default:
-				err := p.doPing()
-				if err != nil {
-					klog.Errorf("Unable to start pinger for IP %q: %v", p.ip, err)
-					return
+				for {
```

Comment:
@yboaron @tpantelis 
I'm probably missing something here. We are now using nested for loops. 
Even if we do not have a for loop in L115, because of the outer for loop at L110, the API (p.doPing) would be executed isn't it? Main difference would be the sleep in L119.

Can you please explain the need for introducing the for loop in Line 115?

Reply:
Okay, we are not returning now when there is an error

Reply:
you're right - we really don't need the inner loop.

Reply:
Yep, you are right.
Deleted the inner loop.
10x